Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.MethodCallExpression.cs

```diff
@@ -63,5 +64,29 @@ public MethodInfo Method
     //     An System.Linq.Expressions.Expression that represents the receiving object
     //     of the method.
     extern public Expression Object { get; }
+
+#if NETFRAMEWORK_4_0 || SILVERLIGHT_4_0 || SILVERLIGHT_5_0
+    //
+    // Summary:
+    //     Creates a new expression that is like this one, but using the supplied children.
+    //     If all of the children are the same, it will return this expression.
+    //
+    // Parameters:
+    //   object:
+    //     The System.Linq.Expressions.MethodCallExpression.Object property of the result.
+    //
+    //   arguments:
+    //     The System.Linq.Expressions.MethodCallExpression.Arguments property of the result.
+    //
+    // Returns:
+    //     This expression if no children are changed or an expression with the updated
+    //     children.
+    [Pure]
+    public MethodCallExpression Update(Expression @object, IEnumerable<Expression> arguments)
+    {
+      Contract.Ensures(Contract.Result<MethodCallExpression>() != null);
```

Comment:
The same is true here (as @sharwell suggested earlier):

```c#
Contract.Requires(Member.IsStatic || expression != null);
Contract.Requires(!Member.IsStatic || expression == null);
```

And from this: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/BinaryExpression.cs,95e70d48bc8fe361

We can add:

`Contract.Requires(!Method.IsGenericMethodDefinition);`
`Contract.Requires(!Method.ContainsGenericParameters);`

:question: What do you think?